django/contrib/postgres/aggregates/mixins.py

```diff
@@ -1,47 +1,24 @@
-from django.core.exceptions import FullResultSet
-from django.db.models.expressions import OrderByList
+import warnings
 
+from django.db.models import Aggregate
+from django.utils.deprecation import RemovedInDjango60Warning
 
-class OrderableAggMixin:
-    def __init__(self, *expressions, ordering=(), **extra):
-        if not ordering:
-            self.order_by = None
-        elif isinstance(ordering, (list, tuple)):
-            self.order_by = OrderByList(*ordering)
-        else:
-            self.order_by = OrderByList(ordering)
-        super().__init__(*expressions, **extra)
-
-    def resolve_expression(self, *args, **kwargs):
-        if self.order_by is not None:
-            self.order_by = self.order_by.resolve_expression(*args, **kwargs)
-        return super().resolve_expression(*args, **kwargs)
-
-    def get_source_expressions(self):
-        return super().get_source_expressions() + [self.order_by]
-
-    def set_source_expressions(self, exprs):
-        *exprs, self.order_by = exprs
-        return super().set_source_expressions(exprs)
 
-    def as_sql(self, compiler, connection):
-        *source_exprs, filtering_expr, ordering_expr = self.get_source_expressions()
+class OrderableAggMixin(Aggregate):
+    allow_order_by = True
 
-        order_by_sql = ""
-        order_by_params = []
-        if ordering_expr is not None:
-            order_by_sql, order_by_params = compiler.compile(ordering_expr)
-
-        filter_params = []
-        if filtering_expr is not None:
-            try:
-                _, filter_params = compiler.compile(filtering_expr)
-            except FullResultSet:
-                pass
-
-        source_params = []
-        for source_expr in source_exprs:
-            source_params += compiler.compile(source_expr)[1]
+    def __init__(self, *expressions, ordering=(), **extra):
+        warnings.warn(
+            "The OrderableAggMixin is deprecated. Use Aggregate and allow_order_by "
+            "instead.",
+            category=RemovedInDjango60Warning,
+        )
```

Comment:
I wonder if we should move this [to `__init_subclass__` instead](https://docs.python.org/3/reference/datamodel.html#object.__init_subclass__) and do something like

```python
class _DeprecatedOrdering:
    def __init__(self, *expressions, ordering=(), **extra):
        if ordering:
            warnings.warn(
                "The ordering argument is deprecated. Use order_by instead.",
                category=RemovedInDjango60Warning,
            )
        super().__init__(*expressions, **extra)

class OrderableAggMixin(_DeprecatedOrdering):
    allow_order_by = True

    def __init_subclass__(cls, /, *args, **kwargs):
        warnings.warn(
            "OrderableAggMixin is deprecated. Use Aggregate and allow_order_by "
            "instead.",
            category=RemovedInDjango60Warning,
        )
        super().__init_subclass__(*args, **kwargs)
```

And then in `postgres.aggregates` subclass `_DeprecatedOrdering` instead and explicitly set `allow_order_by = True` as well.

That would make any subclassing of `OrderableAggMixin` emit the deprecation warning while allowing `postgres.aggregates` to benefit from the deprecation period.